src/modules/31-variables/modals/CreateEditVariableModal/useCreateEditVariableModal.tsx

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright 2021 Harness Inc. All rights reserved.
+ * Use of this source code is governed by the PolyForm Shield 1.0.0 license
+ * that can be found in the licenses directory at the root of this repository, also available at
+ * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
+ */
+
+import React, { useState } from 'react'
+import { useModalHook } from '@harness/use-modal'
+import { Dialog } from '@blueprintjs/core'
+import { useParams } from 'react-router-dom'
+import { useStrings } from 'framework/strings'
+import CreateEditVariable from '@variables/components/CreateEditVariable/CreateEditVariable'
+import type { ProjectPathProps } from '@common/interfaces/RouteInterfaces'
+import type { VariableDTO, VariableRequestDTO } from 'services/cd-ng'
+import css from '@variables/components/CreateEditVariable/CreateEditVariable.module.scss'
+
+export interface UseCreateUpdateVariableModalProps {
+  onSuccess?: ((data: any) => void) | (() => void)
```

Comment:
Codacy has a fix for the issue: [Unexpected any. Specify a different type.](https://app.codacy.com/gh/harness/harness-core-ui/pullRequest?prid=9284062)
```suggestion
  onSuccess?: ((data: unknown) => void) | (() => void)
```